### ISSUE TITLE
Coordinate-based access to matrix.local throws std::out_of_range on unit!=0

### ISSUE THREAD

**Issue by fuerlinger** [dash-project/dash#168]
Using coordinate-based access to the local storage of a matrix throws when not on unit 0. Access works fine using chained subscripts:

````
int main(int argc, char* argv[])
{
  dash::init(&argc, &argv);

  auto myid = dash::myid();
  auto size = dash::size();

  dash::NArray<int, 2> mat(4,10);

  if(myid==0 ) { std::iota(mat.begin(), mat.end(), myid); }
  dash::barrier();

  if( myid==1 ) {
    // cout << mat.local(0,0) << endl; // throws std::out_of_range
    cout << mat.local[0][0] << endl; // works fine
  }

  dash::finalize();
}
````

**Comment by fuchsto** [comment]
Created branch `bug-168-mat-coords`.

**Comment by fuchsto** [comment]
Fixed in #298
